config/routes.rb

```diff
@@ -3,17 +3,27 @@
   # See how all your routes lay out with "rake routes".
 
   # You can have the root of your site routed with "root"
```

Comment:
Your routes look good - I'm glad that you were able to use the /controller/id/action structure for both of the more complex routes